File: tugas/Tugas12_PBO_Adi Sasono 210511017/perpustakaan/Petugas.py
import requests
import json
class Petugas:
    def __init__(self):
        self.__id=None
        self.__code_petugas = None
        self.__nama = None
        self.__jk_petugas = None
        self.__jabatan = None
        self.__no_telepon = None
        self.__alamat = None
        self.__url = "http://localhost/perpustakaan/petugas_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def code_petugas(self):
        return self.__code_petugas
        
    @code_petugas.setter
    def code_petugas(self, value):
        self.__code_petugas = value
    @property
    def nama(self):
        return self.__nama
        
    @nama.setter
    def nama(self, value):
        self.__nama = value
    @property
    def jk_petugas(self):
        return self.__jk_petugas
        
    @jk_petugas.setter
    def jk_petugas(self, value):
        self.__jk_petugas = value
    @property
    def jabatan(self):
        return self.__jabatan
        
    @jabatan.setter
    def jabatan(self, value):
        self.__jabatan = value
    @property
    def no_telepon(self):
        return self.__no_telepon
        
    @no_telepon.setter
    def no_telepon(self, value):
        self.__no_telepon = value
    @property
    def alamat(self):
        return self.__alamat
        
    @alamat.setter
    def alamat(self, value):
        self.__alamat = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_code_petugas(self, code_petugas):
        url = self.__url+"?code_petugas="+code_petugas
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['kode_petugas']
            self.__code_petugas = item['code_petugas']
            self.__nama = item['nama']
            self.__jk_petugas = item['jk_petugas']
            self.__jabatan = item['jabatan']
            self.__no_telepon = item['no_telepon']
            self.__alamat = item['alamat']
        return data
    def simpan(self):
        payload = {
            "code_petugas":self.__code_petugas,
            "nama":self.__nama,
            "jk_petugas":self.__jk_petugas,
            "jabatan":self.__jabatan,
            "no_telepon":self.__no_telepon,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_code_petugas(self, code_petugas):
        url = self.__url+"?code_petugas="+code_petugas
        payload = {
            "code_petugas":self.__code_petugas,
            "nama":self.__nama,
            "jk_petugas":self.__jk_petugas,
            "jabatan":self.__jabatan,
            "no_telepon":self.__no_telepon,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_code_petugas(self,code_petugas):
        url = self.__url+"?code_petugas="+code_petugas
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: tugas/Tugas12_PBO_Adi Sasono 210511017/perpustakaan/Petugas.php
<?php
//Simpanlah dengan nama file : Petugas.php
require_once 'database.php';
class Petugas 
{
    private $db;
    private $table = 'petugas';
    public $code_petugas = "";
    public $nama = "";
    public $jk_petugas = "";
    public $jabatan = "";
    public $no_telepon = "";
    public $alamat = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_code_petugas(int $code_petugas)
    {
        $query = "SELECT * FROM $this->table WHERE code_petugas = $code_petugas";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`code_petugas`,`nama`,`jk_petugas`,`jabatan`,`no_telepon`,`alamat`) VALUES ('$this->code_petugas','$this->nama','$this->jk_petugas','$this->jabatan','$this->no_telepon','$this->alamat')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET code_petugas = '$this->code_petugas', nama = '$this->nama', jk_petugas = '$this->jk_petugas', jabatan = '$this->jabatan', no_telepon = '$this->no_telepon', alamat = '$this->alamat' 
        WHERE kode_petugas = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_code_petugas($code_petugas): int
    {
        $query = "UPDATE $this->table SET code_petugas = '$this->code_petugas', nama = '$this->nama', jk_petugas = '$this->jk_petugas', jabatan = '$this->jabatan', no_telepon = '$this->no_telepon', alamat = '$this->alamat' 
        WHERE code_petugas = $code_petugas";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE kode_petugas = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_code_petugas($code_petugas): int
    {
        $query = "DELETE FROM $this->table WHERE code_petugas = $code_petugas";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>